doc/NAC/PnacSonicHLD.md

```diff
@@ -0,0 +1,374 @@
+Hostapd_wrapper gets notification from hostapd whenever interface authorisation state changes from “Authorised” to “Unauthorised” and from “Unauthorised” to “Authorised”
+
+After receiving interface state change notification from Unauthorised to Authorised hostapd_wrapper performs following two operations,
+- Update column “***nac_status***” in NAC_SESSION table which is part of config_DB to "***authorised***"
+- Update learn_mode for the specific interface in config_db to “***SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW***”  
+This action will trigger Orchagent invoking SAI to update the learn_mode.
+After receiving interface state change notification from Authorised to Unauthorised hostapd_wrapper performs following two operations,
+- Update “***nac_status***” in NAC_SESSION table which is part of config_DB to "***unauthorised***".
+- Update learn_mode for the specific interface in config_db to “***SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DROP***”
+This action will trigger Orchagent invoking SAI to update the learn_mode.
+Invoke ***SAI_API*** to Flush DB for the database
+### 3.2.5 Actions performed when interface operational status changes to DOWN
+Hostapd_wrapper monitors operational status of the interface. When it transitions from “***up***” to “***down***” following actions are triggered by hostapd_wrapper.
+Update learn_mode for the specific interface in config_db to “***SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DROP***”
+Flush FDB for the bridge
+Update column “***nac_status***” in NAC_SESSION table which is part of config_DB to "***unauthorised***"
+Shudown the interface and bringup the interface.
+### 3.2.6 Actions performed when interface operational state changes to UP
+Hostapd_wrapper monitors operational status of the interface. When it transitions from “***down***” to “***up***” no specific actions shall be performed.
+### 3.2.7 Sequence of Events flowchart
+Following Flow chart describes actions in respective sequence after events like port UP/DOWN takes place.
+![PNAC Use Case](images/PNAC_SOE.png)
+
+## 3.3 DB Changes
+As part of PNAC feature introduction following are changes in respective DB.
+### 3.3.1 ConfigDB Schema Changes
+This section describes incremental changes to ConfigDB implemented =for PNAC feature.
+Feature configuration:
+```
+"FEATURE": {
+           "nac": {
+                "auto_restart": "enabled",            
+                "has_global_scope": "True",            
+                "has_per_asic_scope": "False",            
+                "has_timer": "False",            
+                "high_mem_alert": "disabled",            
+                "set_owner": "local",            
+                "state": "enabled"            
+           },
+```        
+***NAC Global configuration***
+```
+"NAC":{
+        "global": {        
+            "admin_state": "up",            
+            "auth_type": "local",            
+            "nac_type": "port"            
+        }        
+    },
+```
+***NAC_Session Table*** 
```

Comment:
Isn't this status of NAC sessions rather than configuration?

Reply:
"NAC_Session_Table" contains both operational status and configuration value maintained per port.